.github/workflows/shippable_builds.yml

```diff
@@ -99,17 +109,81 @@ jobs:
+    outputs:
+      thunderbird_sha: ${{ steps.commit_thunderbird.outputs.thunderbird_sha }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Release Commit Thunderbird
+        id: commit_thunderbird
+        shell: bash
+        env:
+          APP_NAME: ${{ matrix.appName }}
+          PACKAGE_FORMAT: ${{ matrix.packageFormat }}
+          RELEASE_VERSION: ${{ needs.dump_config.outputs.releaseVersion }}
+          RELEASE_TYPE: ${{ needs.dump_config.outputs.releaseType }}
+        run: |
+          git config --global user.name "GitHub Actions Bot"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          if [[ "$APP_NAME" = "thunderbird" && "$PACKAGE_FORMAT" = "apk" ]]; then
+              ./scripts/render-notes.py ${RELEASE_VERSION} ${APP_NAME}
+              # TODO(coreycb): enable once this path is used
+              # git add ./app-metadata/net.thunderbird.android/en-US/changelogs/*
+              git add ./app-metadata/net.thunderbird.android.beta/en-US/changelogs/*
+              git commit -a -m "Release: Thunderbird for Android ${RELEASE_VERSION}"
+              echo "thunderbird_sha=$(git rev-parse HEAD)" | tee >> $GITHUB_OUTPUT
+              git push
+          fi
+
+  commit_k9mail:
+    name: Commit K-9Mail
+    runs-on: ubuntu-latest
+    needs: [commit_thunderbird, dump_config, get_environment]
+    strategy:
+      matrix:
+        include: "${{ fromJSON(needs.dump_config.outputs.matrixInclude) }}"
+    outputs:
+      k9mail_sha: ${{ steps.commit_k9mail.outputs.k9mail_sha }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Release Commit K-9Mail
+        id: commit_k9mail
+        shell: bash
+        env:
+          APP_NAME: ${{ matrix.appName }}
+          RELEASE_VERSION: ${{ needs.dump_config.outputs.releaseVersion }}
+          RELEASE_TYPE: ${{ needs.dump_config.outputs.releaseType }}
+        run: |
+          if [[ "$APP_NAME" = "k9mail" && "$PACKAGE_FORMAT" = "apk" ]]; then
+              ./scripts/render-notes.py ${RELEASE_VERSION} ${APP_NAME}
+              git add ./app-metadata/com.fsck.k9/en-US/changelogs/*
+              git commit -a -m "Release: K-9 Mail ${RELEASE_VERSION}"
+              echo "k9mail_sha=$(git rev-parse HEAD)" | tee >> $GITHUB_OUTPUT
+              git push
+          fi
+
   build_unsigned:
     name: Build Unsigned
     runs-on: ubuntu-latest
     timeout-minutes: 90
-    needs: [dump_config, get_environment]
+    needs: [commit_k9mail, commit_thunderbird, dump_config, get_environment]
     strategy:
       matrix:
         include: "${{ fromJSON(needs.dump_config.outputs.matrixInclude) }}"
     environment: ${{ needs.get_environment.outputs.releaseEnv }}
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
```

Comment:
Something like:

```yaml
- name: Checkout Thunderbird
  if: matrix.appName == 'thunderbird'
  uses: actions/checkout@v4
  with:
    ref: ${{ needs.commit.outputs.thunderbird_sha }}

- name: Checkout Thunderbird
  if: matrix.appName == 'k9mail'
  uses: actions/checkout@v4
  with
    ref: ${{ needs.commit.outputs.k9mail_sha }}
```

or

```yaml
 - name: Checkout repository
  if: matrix.appName == 'thunderbird'
  uses: actions/checkout@v4
  with:
    ref: ${{ matrix.appName == "thunderbird" && needs.commit.outputs.thunderbird_sha || needs.commit.outputs.k9mail_sha }}
```

a third option might be to tag first with the calculated tag name, and then checkout the tag. While this might tag prematurely, we could remove the tag using an `if: failure()` job at the end.

Reply:
Nice, thanks